### PR TITLE
Create control inputs and add them to the documentation

### PR DESCRIPTION
**Pull request by larimaza**
# What

Create control inputs and add them to the documentation in Inputs page.

# Why

Astro received control inputs design specs.

# How

- Create the component styles in `inputs.css`;
- Add the control input icons to the project (these will not go into our icon library);
- Code the logic behind the buttons in `inputs.js`;
- Organize `inputs.js` a little with comments;
- Document the markup in the docs page;
- Change some outline buttons and ghost icon buttons from white to transparent background, so that they'll still look outlined in other context backgrounds other than white (in this case, the disabled input has a grey background);
- Organize `inputs.css` and disable the Stylelint ordering rule, or else we'd have to scatter code related to the same types of input, making maintenance very complicated.

# Sample

Screen Shot 2019-04-30 at 08 18 10

### CODE REVIEW

**Comment by larimaza** [comment]
[Criar componentes de "control input" e inserir na página "Inputs" na documentação](https://trello.com/c/Wjv6iwYz/107-criar-componentes-de-control-input-e-inserir-na-p%C3%A1gina-inputs-na-documenta%C3%A7%C3%A3o)

**Comment by elmarques** [comment]
Guys, I have some few points to consult you:

Captura de Tela 2019-04-30 às 14 17 45

1. When I type a very large number, it gets underneath the buttons.
Do you think it's worth putting padding right in the input to avoid this?

---

Captura de Tela 2019-04-30 às 14 17 52

2. When the disabled field comes with a value, the text goes black.
Could you confirm with the Design team what the expected text color for disabled status, please?

File: src/css/inputs.css [hunk 1]
```diff
@@ -43,22 +47,62 @@
...
+.a-input--ghost.a-input--invalid::after {
+  background-color: var(--color-space-100);
 }
 
-.a-input--invalid > label {
-  color: var(--color-mars-500);
+/* Control inputs */
+
+.a-input--control > input {
+  appearance: textfield;
```
[the remark this answers is not in the document]
Comment (larimaza) [reply src/css/inputs.css]: I didn't either, until research for this PR haha it removes the native increment/decrement buttons in number inputs o/

File: docs/inputs.mdx [hunk 2]
```diff
@@ -112,12 +116,51 @@ This input style does not have a disabled state.
...
+
+To create the disabled state, add the `a-input--disabled` modifier class on element root and `disabled` attribute on the `<input>` tag.
+
+<Playground className='two-columns'>
+  <div className='a-input a-input--control'>
+    <input id='control1' type='number' data-step='100' defaultValue='0' />
+    <label id='controllabel1' htmlFor='control1'>
+      Control input (R$)
+    </label>
+    <button className='a-btn a-btn--ghost-uranus a-btn--small a-btn--icon a-input--control__decrement' >
```
[the remark this answers is not in the document]
Comment (larimaza) [reply docs/inputs.mdx]: Update: we decided to leave it like this for now and revise all of BEM later as a tech debt.

File: docs/js/inputs.js [hunk 3]
```diff
@@ -25,44 +27,107 @@ export const initFloatingLabel = () => {
...
+        currentControlInput.getAttribute('data-step')
+      );
+      const decrementButton = input.querySelector(
+        '.a-input--control__decrement'
+      );
+      const incrementButton = input.querySelector(
+        '.a-input--control__increment'
+      );
+
+      decrementButton.addEventListener('click', () => {
```
[the remark this answers is not in the document]
Comment (larimaza) [reply docs/js/inputs.js]: Perfect! Already tested and done on b570c0c

**Comment by larimaza** [comment]
> 1. **When the disabled field comes with a value, the text goes dark.**
>    Could you confirm with the Design team what the expected text color for disabled status, please?

Great catch! None of the inputs covered that edge case (when filled and disabled), I've added several adjustments to cover all.

**Comment by larimaza** [comment]
> 1. **When I type a very large number, it gets underneath the buttons.**
>    Do you think it's worth putting padding right in the input to avoid this?

Another good catch, also fixed!

**Pull request merged**